lib/vauth/ntlm_wb.c

```diff
@@ -357,6 +357,38 @@ CURLcode Curl_auth_create_ntlm_wb_type1_message(struct Curl_easy *data,
   return result;
 }
 
+/*
+ * Curl_auth_decode_ntlm_wb_type2_message()
+ *
+ * This is used to decode an already encoded NTLM type-2 message. The message
+ * is first decoded from a base64 string into a raw NTLM message and checked
+ * for validity before the appropriate data for creating a type-3 message is
+ * written to the given NTLM data structure.
+ *
+ * Parameters:
+ *
+ * data     [in]     - The session handle.
+ * type2msg [in]     - The base64 encoded type-2 message.
+ * ntlm     [in/out] - The NTLM data struct being used and modified.
+ *
+ * Returns CURLE_OK on success.
+ */
+CURLcode Curl_auth_decode_ntlm_wb_type2_message(struct Curl_easy *data,
+                                                const char *type2msg,
+                                                struct ntlmdata *ntlm)
+{
+  CURLcode result = CURLE_OK;
+
+  (void) data;
```

Comment:
Why specify a `data` argument in a new function when that isn't used?

Reply:
After thinking about it, and as we are not at a stage to only include the internal NTLM code in non-winbind enabled builds, I have reworked the function signatures to only include what arguments are needed.